scrapers/src/scrapers/co/sos/general_candidates.rs

```diff
@@ -28,11 +29,29 @@ impl Scraper {
         html: String,
         context: &crate::ScraperContext<'_>,
     ) -> Result<(), Box<dyn Error>> {
-        let entries = Self::parse_raw_entries(html)?;
-        for entry in entries {
-            let mut office = Self::build_office_input(&entry);
-            office.slug = Some(crate::generate_office_slug(&office));
-            if let Err(err) = db::Office::upsert_from_source(&context.db.connection, &office).await
+        let data = Self::scrape_page_data(html)?;
+
+        let election_year = Self::parse_election_year(&data.title)?;
+        let election_date = crate::generate_general_election_date(election_year)?;
+        let (election_title, election_slug) =
+            crate::generate_general_election_title_slug(&db::State::CO, election_year);
+        let _election = db::Election::upsert_from_source(
+            &context.db.connection,
+            &db::UpsertElectionInput {
+                slug: Some(election_slug),
+                title: Some(election_title),
+                state: Some(db::State::CO),
```

Comment:
For general elections we group all races under a single 'general-election-2028' election with `state: None`.  This is because there are federal races in which everyone is eligible to vote on regardless of where they live and we use the election model as container for all races that fall on a single date.  For the primaries however, there are different dates for different states and those elections can only be voted on by state residents so they are broken out into different elections.

Reply:
ah, that makes sense. Will update 👍